docs/development/standards.rst

```diff
@@ -1,159 +1,75 @@
-        rules in our ``gulpfile.js``. These files should be organized by
-        function and can be split up into multiple files per application.
-
-        Entry points to libraries must be defined in ``gulpfile.js`` for now. We
-        don't have a defined directory structure that would make it easy to
-        imply the entry point to an application library.
-
-.. _`Bower`: http://bower.io
-.. _`Gulp`: http://gulpjs.com
-.. _`Browserify`: http://browserify.org
-.. _`Debowerify`: https://github.com/eugeneware/debowerify
-.. _`LESS`: http://lesscss.org
 
-.. _`jQuery`: http://jquery.com
-.. _`Knockout`: http://knockoutjs.com
+These are build standards that are adhered to by the core development team while
+developing Read the Docs and related services. If you are a new contributor to
+Read the Docs, it might a be a good idea to follow these guidelines as well. The
+:doc:`standard installation instructions <../install>` do cover what it takes to
+get started with a local installation of Read the Docs and can be used for local
+development by non-core team.
+
+Core team standards
+-------------------
+
+Core team members expect to have a development environment that closely
+approximates our production environment, in order to spot bugs and logical
+inconsistencies before they make their way to production.
+
+Celery runs as a separate process
+    Celery needs to be run as a separate process, and core team will not use
+    in-process task execution via Celery's ``ALWAYS_EAGER`` setting. The
+    ``ALWAYS_EAGER`` setting doesn't work correctly, and masks bugs that
+    introduce task execution race conditions.
+
+Celery runs multiple processes
+    We run celery with multiple worker processes to discover race conditions
+    between tasks.
+
+Docker for builds
+    Docker is used for a build backend instead of the local host build backend.
+    There are a number of differences between the two execution methods in how
+    processes are executed, what is installed, and what can potentially leak
+    through and mask bugs -- for example, local SSH agent allowing code check
+    not normally possible.
+
+Setting ``USE_SUBDOMAIN = True``
+    There are a number of resolution bugs and cross-domain behavior that can
+    only be caught by using this setting. It requires use of a catch all public
+    domain, so we can resolve all subdomains to ``localhost``. For example, here
+    are some related settings that affect project URLs::
+
+        PRODUCTION_DOMAIN = 'localhost:8000'
+        PUBLIC_DOMAIN = 'dev.readthedocs.io:8001'
+        USE_SUBDOMAIN = True
+
+Postgres as a database
+    Postgres should be used as the default database whenever possible, as SQLite
+    has issues with our Django version and we use Postgres in production.
+    Differences between Postgres and SQLite should be masked for the most part,
+    as Django does abstract database procedures, and we don't do any
+    Postgres-specific operations yet.
```

Comment:
We glossed over this briefly in the past but I haven't seen a particularly good reason why postgres should be strictly required for development. I get the similarity between dev and prod thing but I'm not sure it outweighs the downsides of having to run a local Postgres instance.

Reply:
There aren't strong reasons to run postgres, which is why this one is more of a recommendation. We're still ultimately running tests using sqlite, so local development and QA against a sqlite db is not likely to uncover any major inconsistencies.

However, postgres is used in production, so therefore you should be comfortable performing maintenance on postgres servers. If using sqlite masks learning about how to run and debug postgres, that's a point against using sqlite.  If we ever decide to use postgres specific queries or features, like native json querying for more performant queries against jsonfield fields, postgres will be a hard requirement.

Postgres is mostly trivial to run locally, and the footprint is minimal. On Linux, i use a psql docker container. On MacOS, is use the postgres application. Setup for either probably matches the time to configure a backported version of sqlite for your system.

I'm fine leaving this as a recommendation

Reply:
> I'm fine leaving this as a recommendation

I agree with this. Although, this point is under "Core team standards" which seems to be mandatory, _but_ it says "whenever possible", so :thinking:

Reply:
It's a strong recommendation :wink:

I can make it sound more optional on this list, but I would like to guide folks to using postgres locally.